gocd/values.yaml

```diff
@@ -298,11 +304,11 @@ agent:
   # postStart:
   #  - "/bin/bash"
   #  - "/agent_startup.sh"
-  # agent.deployStrategy is the strategy explained in detail at https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
   # agent.terminationGracePeriodSeconds is the optional duration in seconds the gocd agent pods need to terminate gracefully.
   # Note: SIGTERM is issued immediately after the pod deletion request is sent. If the pod doesn't terminate, k8s waits for terminationGracePeriodSeconds before issuing SIGKILL.
   # terminationGracePeriodSeconds: 60
-  deployStrategy: {}
+  # agent.controllerStrategy is the strategy explained in detail at https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
+  controllerStrategy: {}
```

Comment:
This is also an unnecessary breaking change for `Deployment`s.

Suggest either
- leave it as `deployStrategy` and make it apply as `strategy` for `Deployment` and `updateStrategy` for `StatefulSet`
- **OR** add a new value of `updateStrategy` that only applies for `StatefulSet`s

Reply:
I'll leave it as `deployStrategy`.